iroh-gateway/src/config.rs

```diff
@@ -167,6 +167,20 @@ impl Source for Config {
     }
 }
 
+impl crate::handlers::StateConfig for Config {
+    fn rpc_client(&self) -> iroh_rpc_client::Config {
+        self.rpc_client.clone()
+    }
+
+    fn port(&self) -> u16 {
+        self.port
+    }
+
+    fn user_headers(&self) -> HeaderMap<HeaderValue> {
```

Comment:
it would be cleaner to return a reference and only `clone` on the caller side if needed